cabal-install/Distribution/Client/Dependency/Types.hs

```diff
@@ -106,9 +113,30 @@ instance Text PreSolver where
       "choose"  -> return Choose
       _         -> Parse.pfail
 
+newtype ReorderGoals = ReorderGoals { unReorderGoals :: Bool }
+  deriving (Eq, Generic, Show)
+
+newtype IndependentGoals = IndependentGoals { unIndependentGoals :: Bool }
+  deriving (Eq, Generic, Show)
+
+newtype AvoidReinstalls = AvoidReinstalls { unAvoidReinstalls :: Bool }
+  deriving (Eq, Generic, Show)
+
+newtype ShadowPkgs = ShadowPkgs { unShadowPkgs :: Bool }
+  deriving (Eq, Generic, Show)
+
+newtype StrongFlags = StrongFlags { unStrongFlags :: Bool }
+  deriving (Eq, Generic, Show)
+
```

Comment:
If all these have an extractor, and `Eq`, `Generic`, and `Show`, shouldn't then `EnableBackjumps` get the same treatment? (Yes, it's an option not exposed to the user, but still ...)